src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs

```diff
@@ -52,7 +52,7 @@ public static partial class JsonSerializer
                 }
 
                 JsonPropertyInfo jsonPropertyInfo = state.Current.JsonClassInfo.GetProperty(propertyName, ref state.Current);
-                if (jsonPropertyInfo == null)
+                if (jsonPropertyInfo == JsonPropertyInfo.s_missingProperty)
```

Comment:
So, what happens if `jsonPropertyInfo` equals null? Or is that guaranteed to never happen now? If so, `Debug.Assert`.

Reply:
Done